src/pytorch_lightning/core/module.py

```diff
@@ -93,7 +93,7 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         torch._C._log_api_usage_once(f"lightning.module.{self.__class__.__name__}")
 
         # pointer to the trainer object
-        self.trainer: Optional["pl.Trainer"] = None
+        self._trainer: Optional["pl.Trainer"] = None
```

Comment:
The idea is that we use the raw attribute if optional checks are desired, the getter otherwise.

Note that some areas of the codebase (saving, model summary, ...) can be used without a Trainer, so they need to use this attribute